src/data_source/storage/pruning.rs

```diff
@@ -0,0 +1,159 @@
+use anyhow::bail;
+use async_trait::async_trait;
+use std::error::Error;
+use std::fmt::Debug;
+use std::time::Duration;
+
+#[derive(Clone, Debug)]
+pub struct PrunerCfg {
+    pruning_threshold: Option<u64>,
+    minimum_retention: Duration,
+    target_retention: Duration,
+    batch_size: u64,
+    max_usage: u16,
+    interval: Duration,
+}
+
+#[async_trait]
+pub trait PruneStorage: PrunerConfig + PrunedHeightStorage {
+    async fn get_disk_usage(&self) -> Result<u64, Self::Error> {
+        Ok(0)
+    }
+
+    async fn prune(&mut self) -> Result<Option<u64>, Self::Error> {
+        Ok(None)
+    }
+
+    fn pruned_height(&self) -> Option<u64> {
```

Comment:
I don't think we use this anymore, do we? Seems like we are always using `load_pruned_height`, the pruned height stored in `NotifyStorage`, or the return value of `prune`.